samples/drivers/adc/adc_sequence/src/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2024 Centro de Inovacao EDGE
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/kernel.h>
+
+/* Macro for building an array of configurations. */
+#define ADC_CHANNEL_CONFIG_DT_AND_COMMA(node_id) ADC_CHANNEL_CFG_DT(node_id),
+
+/* ADC node from the devicetree. */
+#define ADC_NODE DT_ALIAS(adc)
+
+/* Get the ADC resolution from one of its children. */
+#define ADC_RESOLUTION DT_PROP(DT_CHILD(ADC_NODE, channel_0), zephyr_resolution)
```

Comment:
dont like this requirement to name a channel a certain way, even some of the overlays added in this PR are not compliant with this and are not building (like mimxrt1040_evk), just read the resolution from the DT cfg in the main function

Reply:
I think hardcoded is not good because each ADC may not support that resolution, instead make a kconfig and use .conf files for the supported boards of the sample to make sure that a resolution can be used that is supported for each adc

Reply:
Sure thing. Updated to `.conf`.